### PR TITLE
Adding new optional parameter to specify dns servers for Virtual Wan P2SVpnGateway and P2SClients

### PR DESCRIPTION
**Pull request by Nilambari**
## Description
Currently, with VirtualWan-P2SVpnGateway creation, there is no way for customers to specify custom dns servers which they are planning to use. Once, customer can specify custom dns servers, it will get set on gateway(HubVNet) and will be passed to customer Point to site clients as well. So, Point to site clients can also use their dns server/s instead of azure dns. This feature is asked by many customers till now and we have helped them with updating it at backend till this feature releases in PowerShell and customer can add custom dns servers themselves.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by Nilambari** [comment]
Related PowerShell command let command let review :- Azure/azure-powershell-cmdlet-review-pr#585 is approved.
Related merged swagger PR:- Azure/azure-rest-api-specs#9268

ToDo:- Need to add recorded test session file.

File: src/Network/Network/help/New-AzP2sVpnGateway.md [hunk 1]
```diff
@@ -119,11 +120,24 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -CustomDnsServer
+The list of Custom Dns Servers.```yaml
```
Comment (VeryEarly) [inline src/Network/Network/help/New-AzP2sVpnGateway.md:124]: please move ```yaml to next line

File: src/Network/Network/help/Update-AzP2sVpnGateway.md [hunk 2]
```diff
@@ -135,11 +141,24 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -CustomDnsServer
+The list of Custom Dns Servers.```yaml
```
Comment (VeryEarly) [inline src/Network/Network/help/Update-AzP2sVpnGateway.md:145]: please move ```yaml to next line

File: src/Network/Network/help/New-AzP2sVpnGateway.md (hunk 1 above)
Comment (Nilambari) [reply src/Network/Network/help/New-AzP2sVpnGateway.md:124]: Done.

**Comment by Nilambari** [comment]
@VeryEarly Incorporated code review comments and added test record. Can we merge the PR? Thanks!

**Comment by Nilambari** [comment]
> check comments

Please take a look and let me know if merge is fine. Thanks!

**Pull request merged**